src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -218,10 +236,8 @@ public boolean equals(Object other) {
             // state check
             EditPersonDescriptor e = (EditPersonDescriptor) other;
 
-            return getName().equals(e.getName())
-                    && getPhone().equals(e.getPhone())
-                    && getEmail().equals(e.getEmail())
-                    && getAddress().equals(e.getAddress())
+            return getNric().equals(e.getNric()) && getName().equals(e.getName()) && getPhone().equals(e.getPhone())
+                    && getEmail().equals(e.getEmail()) && getAddress().equals(e.getAddress())
```

Comment:
/nit I feel that might me better to follow the original style here to make the code clearer.

Reply:
Final decision: split conditions (one condition one line).